trie/stacktrie.go

```diff
@@ -31,7 +32,12 @@ var (
 
 // StackTrieOptions contains the configured options for manipulating the stackTrie.
 type StackTrieOptions struct {
-	Writer func(path []byte, hash common.Hash, blob []byte) // The function to commit the dirty nodes
+	Writer  func(path []byte, hash common.Hash, blob []byte) // The function to commit the dirty nodes
+	Cleaner func(path []byte)                                // The function to clean up dangling nodes
+
+	SkipLeftBoundary  bool                                             // Flag whether the nodes on the left boundary are skipped for committing
+	SkipRightBoundary bool                                             // Flag whether the nodes on the right boundary are skipped for committing
+	OnBoundary        func(path []byte, hash common.Hash, blob []byte) // Callback invoked when the node is skipped committing
```

Comment:
Do we need this? In this PR, it's only used to `Inc` a meter. Do you have some plans for future usage of it? If we only need it for the meter, we could just pass the gauge and have that here instead.